### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by dev89s**
In this PR:
1. Setup rails with webpack using `rails new -j webpack react-rails-hello`
2. Added `react` & `react-dom` to node-modules using `yarn add react react-dom`
3. Added `@babel/core`, `@babel/preset-env`, `@babel/preset-react`, & `bable-loader` through `yarn add @babel/core @babel/preset-env @babel/preset-react babel-loader --dev`
4. Added Bable config file declaring presets: `@babel/preset-env` & `@babel/preset-react`
5. Added babel loader to `webpack.config.js`:
```
  module: {
    rules: [
      {
        test: /\.(js)$/,
        exclude: /node_modules/,
        use: ['babel-loader'],
      },
    ],
  },
```
7. Added controller for Root of the project: `rails g controller Root index`
8. Configured the routing root to `root 'root#index'`
9. Created and migrated the database 
10. Ran the project with `./bin/dev` and tested to see if it works.
11. Created Redux store and greetings slice
12. Added React router, and made the Root Element
13. Created the backend API for getting a random greeting from the database
14. Created fetch function and an AsyncThunk to fetch and useEffect to dispatch the fetch.
15. Added the greeting to the component view

### CODE REVIEW

File: app/controllers/root_controller.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class RootController < ApplicationController
+  def index; end
+end
```
Comment (mikethreels) [inline app/controllers/root_controller.rb:3]: - [x] according to the requirements there should be a controller and a route for the random greeting as well.

File: db/seeds.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# This file should contain all the record creation needed to seed the database with its default values.
+# The data can then be loaded with the bin/rails db:seed command (or created alongside the database with db:setup).
+#
+# Examples:
+#
+#   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
+#   Character.create(name: "Luke", movie: movies.first)
```
Comment (mikethreels) [inline db/seeds.rb:7]: - [x] in order for this to work there should be a migration file for the greetings table.

File: app/models/application_record.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+class ApplicationRecord < ActiveRecord::Base
+  primary_abstract_class
+end
```
Comment (mikethreels) [inline app/models/application_record.rb:3]: - [x] the same as with the controller and the migration there should be a model to make this work correctly.

File: app/javascript/controllers/application.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { Application } from "@hotwired/stimulus"
+
+const application = Application.start()
+
+// Configure Stimulus development experience
+application.debug = false
+window.Stimulus   = application
+
+export { application }
```
Comment (mikethreels) [inline app/javascript/controllers/application.js]: - [x] According to the requirements, in order for this to work correctly the below list should be implemented:
    - Create the App component with react-router and render it in your static view.
    - Create the Greeting component that will display a greeting. Set it up as a route in your App component.
    - Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
    - Display the random greeting in your Greeting component.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Sasan, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/root_controller.rb (hunk 1 above)
Comment (Whoistolu) [reply app/controllers/root_controller.rb:3]: - Well resolved

File: db/seeds.rb (hunk 2 above)
Comment (Whoistolu) [reply db/seeds.rb:7]: - Well resolved

File: app/models/application_record.rb (hunk 3 above)
Comment (Whoistolu) [reply app/models/application_record.rb:3]: - Well resolved

File: app/javascript/controllers/application.js (hunk 4 above)
Comment (Whoistolu) [reply app/javascript/controllers/application.js]: - Well resolved

**Review by Whoistolu — APPROVED** [review]
### Hi @dev89s 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**